packages/astro/src/vite-plugin-utils/index.ts

```diff
@@ -47,7 +49,8 @@ export function normalizeFilename(filename: string, config: AstroConfig) {
 	if (filename.startsWith('/@fs')) {
 		filename = filename.slice('/@fs'.length);
 	} else if (filename.startsWith('/') && !ancestor(filename, config.root.pathname)) {
-		filename = new URL('.' + filename, config.root).pathname;
+		const url = new URL('.' + filename, config.root);
+		filename = prependForwardSlash(fileURLToPath(url));
```

Comment:
Will this work on Windows? I assume `pathname` was being used here to ensure that we were getting the normalized `/path/to/file` but `fileURLToPath(url)` will give a platform-dependent file path. Maybe the result of that needs to be normalized before the forward slash is prepended?

Reply:
If Windows tests pass, then I digress.

Reply:
Yeah there's something wrong here. We should be using fileURLToPath though, otherwise you get `%20` for the spaces. Something else is wrong.

Reply:
I think `prependForwardSlash(normalizePath(fileURLToPath(url)))` would be correct? Using [`normalizePath`](https://vitejs.dev/guide/api-javascript.html#normalizepath) from Vite.

Reply:
viteID did it. the problem was that fileURLToPath uses back slashes. viteID fixes it.